### ISSUE TITLE
Choosing a GUI library [enhancement]

### ISSUE THREAD

**Issue by benmoran56** [calexil/FightstickDisplay#10]
There are things on the todo list that will require simple GUI functionality. There are a few libraries that work with pyglet.  Two that come to mind are glooey, and simplui. Either one of them should be fine for simple save/load dialogs. 

https://github.com/kxgames/glooey
https://github.com/swiftcoder/simplui

**Comment by calexil** [comment]
ready for the go ahead on this, feel free to pr 

we need to have a configurator that does a button mapping when the player wants, and just run normally else

**Comment by calexil** [comment]
I had a structure in mind for writing out the layout.json but it was wonky

**Comment by benmoran56** [comment]
The layout.json was kinda intended just for moving the graphics around if you make your own theme, but I guess that works too. I have some ideas on how to handle the mapping. 
I'm going to start a new branch with simplui and start playing around.

**Comment by calexil** [comment]
full disclosure, I edited the default layout.json so it would work on my stick, and I have no idea if it works for anything else .lol

I tried for hours to edit the gamecomtroller.py so it would map correctly, until I realized I could just edit the layout.

so, fingers crossed it works for more than just me

**Comment by benmoran56** [comment]
AH!, editing the gamecontroller.py file is definitely the wrong way to do it. Those mappings should already be "perfect" and shouldn't be changed. (They are all taken from the SDL2 database).
The changes should be made in  fightstick.py instead.

For example, pressing the "lefttrigger" button will always fire the "on_trigger_motion" @fightstick.event. 
The question is, what button does this activate on your actual joystick? hk? lk? None? You can change it on lines 141-143 in joystick.py.  (currently, both the right stick AND triggers are activating buttons, but one of them is probably wrong). 

Basically, the events that are triggered should always match what the real button names on your controller. There are a few event types: 
- on_button_press / on_button_release (for a, b, x, y, leftshoulder, rightshoulder, start, select)
- on_stick_motion (for leftstick, rightstick)
- on_dpad_motion (for dpad movement)
- on_trigger_motion (for lefttrigger, righttrigger)
The "stick" and "dpad" ones should be OK already. The other ones need to be fixed up a bit. 

Hope this helps!

**Comment by benmoran56** [comment]
I've started a new branch in my fork with some initial work (based on the simplui module).
It's looking good so far. Not a lot of code, but we already have a simple pop-up menu. I don't think we need a full-blown menu, so something like this is probably plenty. 

https://github.com/benmoran56/FightstickDisplay/tree/test-simplui

**Comment by calexil** [comment]
what im saying is, the button mapping can be solved by simply editing the layout.json.

for instance, the button config code could ask for an input by flashing the button until it's pressed for a couple seconds, then write out to the layout file the correct mapping

**Comment by calexil** [comment]
I am getting started on the writeout tonight
